src/status_im/chat/events.cljs

```diff
@@ -120,38 +110,27 @@
                                  :show-emoji?       false
                                  :bottom-info       details})))
 
+(def ^:private index-messages (partial into {} (map (juxt :message-id identity))))
+
 (handlers/register-handler-fx
   :load-more-messages
   [(re-frame/inject-cofx :get-stored-messages)]
-  (fn [{{:keys [current-chat-id loading-allowed] :as db} :db
+  (fn [{{:keys [current-chat-id] :as db} :db
         get-stored-messages :get-stored-messages} _]
     (let [all-loaded? (get-in db [:chats current-chat-id :all-loaded?])]
-      (if (and loading-allowed (not all-loaded?))
-        (let [messages-path [:chats current-chat-id :messages]
-              messages      (get-in db messages-path)
-              chat-messages (filter #(= current-chat-id (:chat-id %)) messages)
-              new-messages  (get-stored-messages current-chat-id (count chat-messages))
-              all-loaded?   (> const/default-number-of-messages (count new-messages))]
+      (when-not all-loaded?
+        (let [loaded-count (count (get-in db [:chats current-chat-id :messages]))
+              new-messages (get-stored-messages current-chat-id loaded-count)
```

Comment:
have you thought about doing like @flexsurfer and doing this in the co-effect itself so you can effectively deal with values instead of a function to call ?

Reply:
Yes, good idea, and we should do it eventually I just wanted to limit the scope of this PR.

Reply:
Yes, that's definitely better and I will do it later, just wanted to limit the scope of this PR.